packages/iTwinUI-react/src/core/Dialog/DialogTitleBarTitle.tsx

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '@storybook/theming';
+import '@itwin/itwinui-css/css/dialog.css';
+
+export type DialogTitleBarTitleProps = {
+  /**
+   * Dialog title content.
+   */
+  children: React.ReactNode;
+} & React.ComponentPropsWithRef<'div'>;
+
+/**
+ * Dialog title bar. Recommended to be used as a child of `Dialog`.
+ * @example
+ * <Dialog.TitleBar>My dialog title</Dialog.TitleBar>
+ */
+export const DialogTitleBarTitle = React.forwardRef<
+  HTMLDivElement,
+  DialogTitleBarTitleProps
+>((props, ref) => {
+  const { children, className, ...rest } = props;
+  useTheme();
+  return (
+    <div className={cx('iui-dialog-title', className)} ref={ref} {...rest}>
```

Comment:
For supporting multiple elements, we'll still need to improve the CSS to use `margin-right: auto` on this title element instead of `space-between`.

The API won't change but just mentioning this here in case we forget.

Reply:
https://github.com/iTwin/iTwinUI-react/pull/670#discussion_r926312491